sockeye/training.py

```diff
@@ -393,8 +393,12 @@ def _fit(self,
                     checkpoint_state = CheckpointState(checkpoint=train_state.checkpoint, metric_val=m_val)
                     optimizer.pre_update_checkpoint(checkpoint_state)
 
+                # learning rate adjustment
                 if self.lr_scheduler is not None:
-                    self.lr_scheduler.new_evaluation_result(has_improved)
+                    lr_adjusted = self.lr_scheduler.new_evaluation_result(has_improved)
+                    if lr_adjusted and not has_improved:
+                        logger.info("Loading parameters from last best checkpoint: %d", best_checkpoint)
```

Comment:
We could leave ourselves a comment here like:

    # TODO: Try also loading or completely resetting optimizer state